### PR TITLE
Move IAnyDriverError to driver-definitions

### PR DESCRIPTION
**Pull request by jatgarg**
## Description

Move IAnyDriverError to driver-definitions because it needs to be used in layers below loader.

### CODE REVIEW

File: packages/common/driver-definitions/src/driverError.ts [hunk 1]
```diff
@@ -92,6 +92,19 @@ export enum DriverErrorType {
...
+/**
+ * Interface describing errors and warnings raised by any driver code.
+ * Not expected to be implemented by a class or an object literal, but rather used in place of
+ * any or unknown in various function signatures that pass errors around.
+ *
+ * "Any" in the interface name is a nod to the fact that errorType has lost its type constraint.
+ * It will be either DriverErrorType or the specific driver's specialized error type enum,
+ * but we can't reference a specific driver's error type enum in this code.
+ */
+export interface IAnyDriverError extends Omit<IDriverErrorBase, "errorType"> {
```
Comment (vladsud) [inline packages/common/driver-definitions/src/driverError.ts:104]: I wonder if it should be other way around - we define IAnyDriverError interface, and then IDriverErrorBase extends it and overwrites errorType with more concrete type. It should compile without using "Omit" thing

File: packages/loader/driver-utils/src/network.ts [hunk 2]
```diff
@@ -38,8 +38,11 @@ export function isOnline(): OnlineStatus {
  * "Any" in the interface name is a nod to the fact that errorType has lost its type constraint.
  * It will be either DriverErrorType or the specific driver's specialized error type enum,
  * but we can't reference a specific driver's error type enum in this code.
- */
- export interface IAnyDriverError extends Omit<IDriverErrorBase, "errorType"> {
+ *
+ * @deprecated - In favour of {@link @fluidframework/driver-definitions#IAnyDriverError} so that
+ * it can used from the base to upper layers.
+*/
+export interface IAnyDriverError extends Omit<IDriverErrorBase, "errorType"> {
```
Comment (vladsud) [inline packages/loader/driver-utils/src/network.ts:45]: Please open a task to track removal

**Review by vladsud — APPROVED** [review]

File: packages/common/driver-definitions/src/driverError.ts (hunk 1 above)
Comment (jatgarg) [reply packages/common/driver-definitions/src/driverError.ts:104]: It is same thing whatever we do. They both have same structure except the different type in errortype. So we have to use Omit in DriverErrorBase if we go as you suggested. So, I would leave it as such.

File: packages/loader/driver-utils/src/network.ts (hunk 2 above)
Comment (jatgarg) [reply packages/loader/driver-utils/src/network.ts:45]: Created the removal task: https://dev.azure.com/fluidframework/internal/_workitems/edit/2159

**Pull request merged**